### PR TITLE
Revamped modlog with moderation reason

### PR DESCRIPTION
**Pull request by 0xCmdrKeen**
By request, I rebased my old branch for #1046 to hopefully work with the recent code changes.

Had to restore a couple of functions that were removed or changed (`helpers/array.ts#notEmpty()` and `useIsAdmin()` with optional argument). Only tested briefly but seems to work fine.

### CODE REVIEW

File: src/helpers/array.ts
Comment (aeharding) [inline src/helpers/array.ts:3]: You can use lodash compact on the array instead.

File: src/features/moderation/logs/ModlogItem.tsx
Comment (aeharding) [inline src/features/moderation/logs/ModlogItem.tsx:136]: Why were these `as` added? Typescript should be able to infer this

**Review by aeharding — COMMENTED** [review]
Thanks for making this PR, I'll review it in detail soon!

File: src/helpers/array.ts
Comment (0xCmdrKeen) [reply src/helpers/array.ts:3]: Oh gotcha. Okay, I'll fix it.

File: src/features/moderation/logs/ModlogItem.tsx
Comment (0xCmdrKeen) [reply src/features/moderation/logs/ModlogItem.tsx:136]: I made this months ago, it's possible that VS Code was giving me warnings, but perhaps that's changed. I'll review.

Comment (0xCmdrKeen) [reply src/features/moderation/logs/ModlogItem.tsx]: Yep, looks like it's no longer necessary. Removed.

File: src/features/moderation/useIsAdmin.tsx
Comment (aeharding) [inline src/features/moderation/useIsAdmin.tsx:4]: What is the purpose of this change?

File: src/helpers/useAppNavigation.ts
Comment (aeharding) [inline src/helpers/useAppNavigation.ts:53]: Can you use `buildCommunityLink` from `appLinkBuilder`? Like `buildGeneralBrowseLink(buildCommunityLink(community))`.

(Sorry I know it's not done everywhere, I'm slowly switching over to it.)

You can make `buildCommunityLink` take a `CommunityView` if it would help!

File: src/features/moderation/logs/ModlogItem.tsx
Comment (aeharding) [inline src/features/moderation/logs/ModlogItem.tsx:195]: I see you're using the styles from InboxItem, which is great! Do you think its worth abstracting them into a component that takes `icon`? Perhaps for another task?

File: src/features/moderation/useIsAdmin.tsx
Comment (0xCmdrKeen) [reply src/features/moderation/useIsAdmin.tsx:4]: I needed a way to determine admin status for arbitrary users so I can properly highlight them in the modlog (see ModLogItem.tsx:171). 

This is basically the previous version of this function (from before it was replaced with a static selector in ead60d3), augmented by an optional `person` parameter.

Since it's not used anywhere else right now, I can make this a private function in `ModLogItem` instead and restore this helper to its last state if you prefer.

File: src/helpers/useAppNavigation.ts
Comment (0xCmdrKeen) [reply src/helpers/useAppNavigation.ts:53]: Sure, no problem.

File: src/features/moderation/logs/ModlogItem.tsx
Comment (0xCmdrKeen) [reply src/features/moderation/logs/ModlogItem.tsx:195]: Good question. I did indeed copy some of the styled components from there in order to have the design stay consistent across the app, thinking it would make sense to have the modlog look similar to a view that users will already be familiar with, that way it would be easier to read.

The reason I didn't extract them into shared components is because they are still separate features and I didn't want to unecessarily entangle them in case they evolve somewhat differently in the future.

File: src/features/moderation/useIsAdmin.tsx
Comment (aeharding) [reply src/features/moderation/useIsAdmin.tsx:4]: I see. How about something like `useAppSelector(!person ? isAdminSelector : state => state.site.response?.admins?.some(
      (admin) => admin.person.id === person?.id
    ))` instead?

Comment (0xCmdrKeen) [reply src/features/moderation/useIsAdmin.tsx:4]: Sounds good, can do.

**Review by aeharding — APPROVED** [review]
Looks good, just failing CI

**Comment by aeharding** [comment]
DOH, I don't know why Github didn't wait for successful CI before merging :(

**Pull request merged**
